src/crimson/common/fatal_signal.cc

```diff
@@ -0,0 +1,78 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "fatal_signal.h"
+
+#include <csignal>
+#include <iostream>
+#include <string_view>
+
+#define BOOST_STACKTRACE_USE_ADDR2LINE
+#include <boost/stacktrace.hpp>
+#include <seastar/core/reactor.hh>
+
+FatalSignal::FatalSignal()
+{
+  install_oneshot_signals_handler<SIGSEGV,
+                                  SIGABRT,
+                                  SIGBUS,
+                                  SIGILL,
+                                  SIGFPE,
+                                  SIGXCPU,
+                                  SIGXFSZ,
+                                  SIGSYS>();
+}
+
+template <int... SigNums>
+void FatalSignal::install_oneshot_signals_handler()
+{
+  (install_oneshot_signal_handler<SigNums>() , ...);
+}
+
+template <int SigNum>
+void FatalSignal::install_oneshot_signal_handler()
+{
+  struct sigaction sa;
+  sa.sa_sigaction = [](int sig, siginfo_t *info, void *p) {
+    if (static std::atomic_bool handled{false}; handled.exchange(true)) {
+      return;
+    }
+    FatalSignal::signaled(sig);
+    ::signal(sig, SIG_DFL);
+  };
+  sigfillset(&sa.sa_mask);
+  sa.sa_flags = SA_SIGINFO | SA_RESTART;
+  if constexpr (SigNum == SIGSEGV) {
+    sa.sa_flags |= SA_ONSTACK;
+  }
+  [[maybe_unused]] auto r = ::sigaction(SigNum, &sa, nullptr);
+  assert(r);
```

Comment:
@rzarzynski Is this actually correct?  The man page seems to say that 0 means success?

Reply:
Oops, I missed the negation here. The problem slipped in undetected b/c using productions builds.
Fixing. Thanks for pointing this out!

Reply:
https://github.com/ceph/ceph/pull/41010

Reply:
thank you for your prompt fix. should have been more carefully when reviewing this change. =(